test/device/utils.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <gtest/gtest.h>
+
+#include "../../source/device.hpp"
+#include "../../source/link.hpp"
+
+namespace sim {
```

Comment:
I think it would be better to use other namespace:
```suggestion
namespace test {
```
Code will become more clear (in places you creates mocks you will see that this objects are from `test` namespace). Also you will have no able to accidentally use them in main project